src/Microsoft.Azure.WebJobs.Host/Extensions/IJobHostMetadataProvider.cs

```diff
@@ -60,5 +60,12 @@ public interface IJobHostMetadataProvider
         /// </summary>
         /// <param name="output">The <see cref="TextWriter"/> to write the graph to.</param>
         void DebugDumpGraph(TextWriter output);
+
+        /// <summary>
+        /// Gets function metadata.
```

Comment:
Rather than MethodMetadata, I think we should name the type FunctionMetadata - we use the "Function" terminology elsewhere. Note that you used "function" in your xml description as well because it is more natural :)

So You'd have `FunctionMetadata GetFunctionMetadata(string functionName)`

Reply:
Though I do see that in the functions runtime, we already have a public type "FunctionMetadata". Likely our usage of the new core SDK API won't cause any conflicts because it is constrained, and namespacing can resolve any conflicts.